### PR TITLE
Spring Component Scanning

### PR DESCRIPTION
**Pull request by itsLucario**
Please re-read your article a day or so after you finished writing it and work through this checklist to polish the quality. This will save all of us a lot of time during review.

## Content
- [ ] the article explains *why* and not only *how*
- [ ] (half-)sentences that express a meaningful idea that is understandable by itself own are **bold** to allow quick scanning of the text

## Text
- [x] the text is conversational ("we" instead of "you", easy language)
- [x] there are no "walls of text" (short sentences, short paragraphs with a single idea)
- [ ] the text is inclusive ("the developers" and "they" instead of "the developer" and "he/she")
- [x] the text uses active voice instead of passive voice
- [x] names are spelled consistently throughout the article (correct uppercase / lowercase)
- [x] file names and variables are highlighted as `code`

## Structure
- [x] the article has a short intro before the first headline
- [ ] by reading the table of contents (TOC) alone, I get a sense of the "story" of the article (check the TOC in the preview after raising the PR or by starting the blog locally on your machine)
- [ ] if using sub-headlines, there's at least two sub-headlines below a main-headline
- [ ] headlines are consistent throughout the article (same level of abstraction, same style - if one headline starts with a verb in imperative form, the other headlines on the same level should probably, too)

## Code Examples
- [ ] code examples are introduced with a ":" and explained below the code example
- [x] code examples are formatted to avoid horizontal scrolling (manual line breaks and 2-spaces indentation)
- [x] the text has been checked with [Grammarly](https://grammarly.com) (the free tier is enough)

**Feel free to raise the PR even if the checklist hasn't been worked through, yet**, to take advantage of the preview that is generated for each pull request (see "deploy/netlify" under "Show all checks"). But please only flag it to be reviewed after going through the checklist above.

### CODE REVIEW

File: _posts/2020-09-31-spring-component-scanning.md [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+    }
+  }
+}
+```
+
+We'll be using `ApplicationContext`'s `getBeanDefinitionNames()` method throughout this article to check out the list of beans created under the application context.
+
+The above `BeanViewer` will print all the beans that are registered with the application context. This will help us to check whether our components are loaded properly or not.
+
+### Spring Boot's Implicit Auto Scanning
```
Comment (thombergs) [inline _posts/2020-09-31-spring-component-scanning.md:90]: This section shows that Spring Boot's auto scanning works, but not why it works. Please add a code example that shows the code of the `@SpringBootApplication` annotation and explain how it is meta-annotated with `@ComponentScan`.

File: _posts/2020-09-31-spring-component-scanning.md [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+...
+```
+
+The above output shows the bean created for `BeanViewer`, `ExplicitScan`, and `Users` and printed by our `BeanViewer`. Since all these classes come under our parent package `io.reflectoring.componentscan` these are auto scanned by Spring.
+
+### Using `@ComponentScan` without any attributes
+
+If we have a package that does not come under our parent package, we can use `@ComponentScan` along with `@Configuration`. This will tell Spring to scan the components in the package where this `@Configuration` class is present and its sub-packages.
+
+```java
```
Comment (thombergs) [inline _posts/2020-09-31-spring-component-scanning.md:120]: That's three code examples in a row without any "story" in between. Please have a look at how I changed this to make it more readable embedded in a "story".

**Review by thombergs — CHANGES_REQUESTED** [review]
The content was quite good, but the order of things and some explanations were hard to follow, so I edited it quite heavily. Please have a look at the changes and the open comments.

File: _posts/2020-09-31-spring-component-scanning.md (hunk 2 above)
Comment (itsLucario) [reply _posts/2020-09-31-spring-component-scanning.md:120]: Yeah got it @thombergs. I'll work on my explanations.
Sorry for making you to go through lot of modifications.

File: _posts/2020-09-31-spring-component-scanning.md (hunk 1 above)
Comment (itsLucario) [reply _posts/2020-09-31-spring-component-scanning.md:90]: Actually I have explained that in "When to Use Component Scanning Section"
--------------------
If the application is a Spring Boot application, then all the packages under the package containing the Spring Boot application class will be covered by an implicit component scan. 

Spring Boot's `@SpringBootApplication` annotation implies the `@Configuration`, `@ComponentScan`, and `@EnableAutoConfiguration` annotations.
